packages/lib/models/Note.ts

```diff
@@ -828,9 +828,15 @@ export default class Note extends BaseItem {
 		void ItemChange.add(BaseModel.TYPE_NOTE, savedNote.id, isNew ? ItemChange.TYPE_CREATE : ItemChange.TYPE_UPDATE, changeSource, beforeNoteJson);
 
 		if (dispatchUpdateAction) {
+			// The UI requires share_id -- if a new note, it will always be the empty string in the database
+			// until processed by the share service. At present, loading savedNote from the database in this case
+			// breaks tests.
+			if (!('share_id' in savedNote) && isNew) {
+				savedNote.share_id = '';
+			}
 			// Ensures that any note added to the state has all the required
 			// properties for the UI to work.
-			if (!('deleted_time' in savedNote)) {
+			if (!('deleted_time' in savedNote) || !('share_id' in savedNote)) {
```

Comment:
> `			if (!('deleted_time' in savedNote) || !('share_id' in savedNote)) {`

This particular change (adding `|| !('share_id' ...`) may not be necessary to fix the originally reported issue (which only applies to new notes). However, because `share_id` not being present in `state.notes` can cause crashes, I think it makes sense to ensure that `share_id` should be present in both new and existing notes.